### PR TITLE
build: Add CMake build system for C Library

### PR DESCRIPTION
**Pull request by mtm0183**
ci: Adjust gradle build scripts for C Library building
feat: Add initial support for ATECC508A, ATECC608A and mbed TLS
test: Add test for ATECC508A

Signed-off-by: Mark Mulrooney <mark@ockam.io>

Thank you for sending a pull request :heart:

### Checklist

Please review the [guidelines for contributing code](../CONTRIBUTING.md#contribute-code) to this repository. And check `[x]` all the boxes that apply:

- [ ] This request **pulls a feature/bugfix branch** (right side) from my fork. *Not my master.*
- [ ] This request **pulls against** `ockam-network/ockam`’s **develop branch** (left side).
- [ ] Build succeeds `./build` with no linter warnings or test failures.
- [ ] All code matched our code formatting [conventions](../CONTRIBUTING.md#code-format).
- [ ] Commit messages match our [conventions](../CONTRIBUTING.md#commit-messages) and all commits
are [signed](#signed-commits).
- [ ] This request includes tests and documentation for all new code.

### Proposed Changes
Please describe the changes in your pull request.

If this pull request resolves an already recorded bug or a feature request, be sure to link to that issue.

### CODE REVIEW

File: implementations/c/config/vault/ockam_vault_hw_cfg.h [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+ ********************************************************************************************************
+ *                                           GLOBAL FUNCTIONS                                           *
+ ********************************************************************************************************
+ */
+
+/*
+ ********************************************************************************************************
+ *                                            LOCAL FUNCTIONS                                           *
+ ********************************************************************************************************
+ */
```
Comment (mrinalwadhwa) [inline implementations/c/config/vault/ockam_vault_hw_cfg.h]: remove these place holder comments from all files, we can add them wherever we really need them

File: implementations/c/config/vault/ockam_vault_hw_cfg.h [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+
+#ifndef OCKAM_VAULT_HW_CFG_H_
+#define OCKAM_VAULT_HW_CFG_H_
+
+/*
+ ********************************************************************************************************
+ *                                             INCLUDE FILES                                            *
+ ********************************************************************************************************
+ */
+
```
Comment (mrinalwadhwa) [inline implementations/c/config/vault/ockam_vault_hw_cfg.h]: do we really need this header? doesn't a series of `#include` statements convey the same thing.

File: implementations/c/config/vault/ockam_vault_hw_cfg.h [hunk 3]
```diff
@@ -0,0 +1,69 @@
...
+#include <common/include/ockam_def.h>
+
+
+/*
+ ********************************************************************************************************
+ *                                                DEFINES                                               *
+ ********************************************************************************************************
+ */
```
Comment (mrinalwadhwa) [inline implementations/c/config/vault/ockam_vault_hw_cfg.h]: do we really need this header? doesn't a series of `#define` statements convey the same thing.

File: implementations/c/include/kal/ockam_kal.h [hunk 4]
```diff
@@ -0,0 +1,155 @@
...
+ *                                               MUTEX                                                  *
+ ********************************************************************************************************
+ */
+
+OCKAM_ERR  ockam_kal_mutex_init (OCKAM_KAL_MUTEX *p_mutex);
+
+OCKAM_ERR  ockam_kal_mutex_free (OCKAM_KAL_MUTEX *p_mutex);
+
+OCKAM_ERR  ockam_kal_mutex_lock (OCKAM_KAL_MUTEX *p_mutex,
+                                   OCKAM_KAL_OPT opt,
```
Comment (mrinalwadhwa) [inline implementations/c/include/kal/ockam_kal.h]: broken indent.

File: implementations/c/include/kal/ockam_kal.h [hunk 5]
```diff
@@ -0,0 +1,155 @@
...
+OCKAM_ERR  ockam_kal_mutex_init (OCKAM_KAL_MUTEX *p_mutex);
+
+OCKAM_ERR  ockam_kal_mutex_free (OCKAM_KAL_MUTEX *p_mutex);
+
+OCKAM_ERR  ockam_kal_mutex_lock (OCKAM_KAL_MUTEX *p_mutex,
+                                   OCKAM_KAL_OPT opt, 
+                                   uint32_t timeout_ms);
+
+OCKAM_ERR  ockam_kal_mutex_unlock (OCKAM_KAL_MUTEX *p_mutex,
+                                  OCKAM_KAL_OPT opt);
```
Comment (mrinalwadhwa) [inline implementations/c/include/kal/ockam_kal.h]: inconsistent indent.

File: .gitmodules [hunk 6]
```diff
@@ -0,0 +1,10 @@
+[submodule "implementations/c/third-party/microchip/cryptoauthlib"]
```
Comment (mrinalwadhwa) [inline .gitmodules]: This, I suspect, is unsafe.
This build will break if the third-party commits a breaking change to their master.

We should move close a pinned commit of the repo to a gradle task.

Comment (mtm0183) [reply .gitmodules]: We're still in a heavy development phase. If they push an update I'd rather get it, have it break the build and understand what changed rather then lock us into one commit at the moment.

File: implementations/c/config/vault/ockam_vault_hw_cfg.h (hunk 1 above)
Comment (mtm0183) [reply implementations/c/config/vault/ockam_vault_hw_cfg.h]: I'd disagree. Having the comment blocks here ensure code is written in a consistent fashion. In ever file we know where the global variables, data types, etc. are located.

File: implementations/c/config/vault/ockam_vault_hw_cfg.h (hunk 2 above)
Comment (mtm0183) [reply implementations/c/config/vault/ockam_vault_hw_cfg.h]: It ensures consistency of include location across all files.

File: implementations/c/config/vault/ockam_vault_hw_cfg.h (hunk 3 above)
Comment (mtm0183) [reply implementations/c/config/vault/ockam_vault_hw_cfg.h]: It ensures consistency of include location across all files.

File: implementations/c/include/kal/ockam_kal.h [hunk 7]
```diff
@@ -0,0 +1,155 @@
...
+
+/**
+ *******************************************************************************
+ * @enum    OCKAM_KAL_OPT
+ * @brief   Options when making OS calls
+ *******************************************************************************
+ */
+
+typedef enum {
+    OCKAM_KAL_OPT_NONE              = 0x00,                     /*!< No option specified                                */
```
[the remark this answers is not in the document]
Comment (mtm0183) [reply implementations/c/include/kal/ockam_kal.h]: What's the advantage of 120? I typically go out to 122.

File: implementations/c/build.gradle [hunk 8]
```diff
@@ -4,41 +4,24 @@ plugins {
...
-      script '''
-        mkdir -p _build/x86_64-unknown-linux-gnu
-        cd _build/x86_64-unknown-linux-gnu
-        cmake ../.. -DCMAKE_INSTALL_PREFIX=./_install
-        make
-        make install
-      '''
-    }
-  }
+  dependsOn getDependencies, gradle.includedBuild('testatecc508a').task(':build')
```
[the remark this answers is not in the document]
Comment (mtm0183) [reply implementations/c/build.gradle]: This code proceeds the above line:
`onlyIf { host.debianBuilder.enabled }`
Is this what you're asking for?

File: .editorconfig [hunk 9]
```diff
@@ -14,8 +14,15 @@ indent_style = space
...
 
+
+[*.c, *.h]
+indent_style = space
+indent_size = 4
+
+
 [Makefile]
-indent_style = tab
+indent_style = space
```
Comment (mrinalwadhwa) [inline .editorconfig]: I suspect Makefile with a space indent will throw an error. Make only allows tabs

**Pull request merged**